language/vm/vm-runtime/vm-runtime-types/src/native_functions/signature.rs

```diff
@@ -110,7 +110,13 @@ pub fn native_ed25519_threshold_signature_verification(
 }
 
 fn ed25519_threshold_signature_verification_cost(num_of_sigs: u64, message_len: usize) -> u64 {
-    BATCH_ED25519_COST * num_of_sigs * message_len as u64
+    BATCH_ED25519_COST
+        .checked_mul(
+            num_of_sigs
+                .checked_mul(message_len as u64)
+                .expect("no overflow"),
+        )
+        .expect("no overflow")
```

Comment:
I looked over the code and couldn't find a simple explanation for arguing that an overflow can never happen, so perhaps it is prudent to insert some runtime checks here.

Reply:
cc @kchalkias